core/arch/arm/plat-imx/registers/imx7-gpc_regs.h

```diff
@@ -0,0 +1,353 @@
+#define BP_GPC_PGC_ACK_SEL_A7_A7_C1_PGC_PDN_ACK		(1)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_C1_PGC_PDN_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_C1_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_A7_PLAT_PGC_PDN_ACK	(2)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_PLAT_PGC_PDN_ACK	\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_PLAT_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_MF_PGC_PDN_ACK		(3)
+#define BM_GPC_PGC_ACK_SEL_A7_MF_PGC_PDN_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_MF_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_MIPI_PGC_PDN_ACK		(4)
+#define BM_GPC_PGC_ACK_SEL_A7_MIPI_PGC_PDN_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_MIPI_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_PCIE_PGC_PDN_ACK		(5)
+#define BM_GPC_PGC_ACK_SEL_A7_PCIE_PGC_PDN_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_PCIE_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_USB_OTG1_PGC_PDN_ACK	(6)
+#define BM_GPC_PGC_ACK_SEL_A7_USB_OTG1_PGC_PDN_ACK	\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_USB_OTG1_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_USB_OTG2_PGC_PDN_ACK	(7)
+#define BM_GPC_PGC_ACK_SEL_A7_USB_OTG2_PGC_PDN_ACK	\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_USB_OTG2_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_HSIC_PGC_PDN_ACK		(8)
+#define BM_GPC_PGC_ACK_SEL_A7_HSIC_PGC_PDN_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_HSIC_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_A7_PGC_PDN_ACK		(15)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_PGC_PDN_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_PGC_PDN_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_A7_C0_PGC_PUP_ACK		(16)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_C0_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_C0_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_A7_C1_PGC_PUP_ACK		(17)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_C1_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_C1_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_A7_PLAT_PGC_PUP_ACK	(18)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_PLAT_PGC_PUP_ACK	\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_PLAT_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_MF_PGC_PUP_ACK		(19)
+#define BM_GPC_PGC_ACK_SEL_A7_MF_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_MF_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_MIPI_PGC_PUP_ACK		(20)
+#define BM_GPC_PGC_ACK_SEL_A7_MIPI_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_MIPI_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_PCIE_PGC_PUP_ACK		(21)
+#define BM_GPC_PGC_ACK_SEL_A7_PCIE_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_PCIE_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_USB_OTG1_PGC_PUP_ACK	(22)
+#define BM_GPC_PGC_ACK_SEL_A7_USB_OTG1_PGC_PUP_ACK	\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_USB_OTG1_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_USB_OTG2_PGC_PUP_ACK	(23)
+#define BM_GPC_PGC_ACK_SEL_A7_USB_OTG2_PGC_PUP_ACK	\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_USB_OTG2_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_HSIC_PGC_PUP_ACK		(24)
+#define BM_GPC_PGC_ACK_SEL_A7_HSIC_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_HSIC_PGC_PUP_ACK)
+#define BP_GPC_PGC_ACK_SEL_A7_A7_PGC_PUP_ACK		(31)
+#define BM_GPC_PGC_ACK_SEL_A7_A7_PGC_PUP_ACK		\
+			(1 << BP_GPC_PGC_ACK_SEL_A7_A7_PGC_PUP_ACK)
```

Comment:
This case is one of the primary reasons why we have the `BIT*()` macros since `1 << 31` is undefined behavior.

Reply:
I understand, We are using those include files in other projects . This is why we kept it as-is. 
We will rework them for the OP-TEE project then.